src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -372,7 +372,10 @@ private static ImmutableArray<ISymbol> GetSymbolInfo(SemanticModel semanticModel
             }
         }
 
-        return semanticModel.GetSymbolInfo(node, cancellationToken).GetBestOrAllSymbols();
+        var preprocessingSymbol = semanticModel.GetPreprocessingSymbolInfo(node).Symbol;
+        return preprocessingSymbol != null
+            ? ImmutableArray.Create<ISymbol>(preprocessingSymbol)
+            : semanticModel.GetSymbolInfo(node, cancellationToken).GetBestOrAllSymbols();
```

Comment:
We enable a common path for the retrieval of the preprocessing symbol to use in the reference finder. This avoids specifically handling preprocessing symbols when querying the semantic model for finding its references in the finder.

Reply:
> This avoids specifically handling preprocessing symbols when querying the semantic model for finding its references in the finder.

Not sure what this means.

Reply:
Invalid comment, this is part of the actual implementation and now uses the custom preprocessing symbol method in the semantic facts instead of querying the semantic model.